src/MSALWrapper.Test/GlobalLockTest.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.MSALWrapper.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Authentication.MSALWrapper;
+    using NUnit.Framework;
+
+    /// <summary>
+    /// The <see cref="GlobalLockTest"/> test.
+    /// </summary>
+    internal class GlobalLockTest
+    {
+        /// <summary>
+        /// The number of threads or tasks.
+        /// </summary>
+        private const int NumberOfThreads = 10;
+
+        /// <summary>
+        /// The simulated lockname.
+        /// </summary>
+        private string lockName;
+
+        /// <summary>
+        /// The setup.
+        /// </summary>
+        [SetUp]
+        public void Setup()
+        {
+            this.lockName = Guid.NewGuid().ToString();
+        }
+
+        /// <summary>
+        /// Test mutex in single task.
+        /// </summary>
+        [Test]
+        public void TestSingleMutex()
+        {
+            int sum = 0;
+            Task.Run(() =>
+            {
+                using (new GlobalLock(this.lockName))
+                {
+                    sum++;
+                }
+            }).Wait();
+            Assert.AreEqual(1, sum);
+        }
+
+        /// <summary>
+        /// Test mutex in tasks.
+        /// </summary>
+        [Test]
+        public void TestMutexInTasks()
+        {
+            int cas = 0;

```

Comment:
It's not clear to me on first sight, what the variable name `cas` means. Consider a more description name.

Reply:
used semaphore instead